src/components/KeywordsDistribution/KeywordsDistribution.tsx

```diff
@@ -0,0 +1,225 @@
+
+const diagramColors = [
+  'cyan',
+  'orange',
+  'green',
+  'red',
+  'magenta',
+  'gray',
+  'brown',
+  'blue',
+  'yellow',
+  'white',
+];
+const diagramColorsLength = diagramColors.length;
+
+const minSliderDistance = 3600000;
+const xDelta = 86400000;
+const yDelta = 1;
+
+export const KeywordsDistribution: FC<KeywordsDistributionProps> = ({ data }) => {
+  const [linearData, barData, domainX, domainY, words] = useMemo(() => {
+    try {
+      return prepareData(data);
+    } catch (e) {
+      toast.error((e as Error).message);
+      return [[], [], [0, 1] as Range, [0, 1] as Range, []];
+    }
+  }, [data]);
+  const [hidden, setHidden] = useState<number[]>([]);
+  const [type, setType] = useState<KeywordsDistributionType>(KeywordsDistributionType.linear);
+
+  const [xRange, setXRange] = useState<Range>(domainX);
+  const [linear, setLinear] = useState<LinearData>(linearData);
+  const [bar, setBar] = useState<BarData>(barData);
+
+  useEffect(() => setHidden([]), [data]);
+
+  useEffect(() => {
+    setLinear(linearData.map(item => ({
+      ...item,
+      data: item.data.filter(({ unixtime }) => (xRange[0] <= unixtime) && (unixtime <= xRange[1])),
+    })));
+  }, [linearData, xRange, domainY]);
+
+  useEffect(() => {
+    setBar(barData.filter(({ unixtime }) => (xRange[0] <= unixtime) && (unixtime <= xRange[1])));
+  }, [barData, xRange, domainY]);
+
+  const toggleHidden =(e: SyntheticEvent, idx: number) => {
+    if (!(e.target as HTMLInputElement).checked) {
+      setHidden(prev => [...prev, idx]);
+    } else {
+      setHidden(prev => prev.filter(item => item !== idx));
+    }
+  };
+
+  const toggleType = (type: KeywordsDistributionType) => setType(type);
+
+  const handleSlider = useCallback((event: Event, newValue: number | number[], activeThumb: number) => {
+    if (!Array.isArray(newValue)) {
+      return;
+    }
+    if (newValue[1] - newValue[0] < minSliderDistance) {
+      if (activeThumb === 0) {
+        const clamped = Math.min(newValue[0], domainX[1] - minSliderDistance);
+        setXRange([clamped, clamped + minSliderDistance]);
+      } else {
+        const clamped = Math.max(newValue[1], domainX[0] + minSliderDistance);
+        setXRange([clamped - minSliderDistance, clamped]);
+      }
+    } else {
+      setXRange(newValue as Range);
+    }
+  }, [domainX]);
+
+  return (
+    <div className="h-full flex">
+      <div className="h-full flex-1 pb-16 relative">
+        {(type === KeywordsDistributionType.linear) && (
+          <ResponsiveContainer>
+            <LineChart>
+              <CartesianGrid strokeDasharray="3 3" />
+              <XAxis dataKey="unixtime" type="number" domain={xRange} tickFormatter={unixTimeFormatter} interval="preserveStartEnd" angle={-15} />
+              <YAxis dataKey="value" domain={domainY} />
+              <Tooltip contentStyle={tooltipContentStyle} labelStyle={tooltipLabelStyle} labelFormatter={unixTimeFormatter} />
+              {linear.map((s, idx) => (
```

Comment:
I guess here would be better to filter `linear` array before mapping it to `Line` components
I see you've replaced filtering with usage of the `Line` component `hide` prop. But user will see `Tooltip` with info of non-zero occurrences of keyword even if corresponding `Line` is hidden
I attach discribed scenario
![issue](https://user-images.githubusercontent.com/42252206/221672825-9d094393-ebe0-448f-8e28-8f069e5d5e52.gif)
By the way, `Bar` component works fine even without filtering

Reply:
done